packages/read-model/src/Product/PriceFactory.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\ReadModelBundle\Product;
+
+use Shopsys\FrameworkBundle\Component\Money\Money;
+use Shopsys\FrameworkBundle\Model\Pricing\Group\PricingGroup;
+use Shopsys\FrameworkBundle\Model\Pricing\Price;
+use Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPrice;
+
+class PriceFactory
+{
+    /**
+     * @param array $pricesArray
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Group\PricingGroup $pricingGroup
+     * @return \Shopsys\FrameworkBundle\Model\Product\Pricing\ProductPrice|null
+     */
+    public static function createProductPriceFromArrayByPricingGroup(
```

Comment:
Yes.

It can be static, so why not to made it static. Doing so allows me to use it in `ProductDetailViewElasticsearchFactory` without passing dependency od `PriceFactory`.

Reply:
Yeah but it disables ability to extend that method from project-base

Reply:
I do not think so. Or maybe I am not geting your point.

Following example works as expected:
```php
<?php

class A
{
    public static function foo(): string
    {
        return 'original';
    }
}

class B extends A
{
    public static function foo(): string
    {
        return 'replacement';
    }
}

var_dump(A::foo()); // "original"
var_dump(B::foo()); // "replacement"

```

Reply:
Yes, but cannot change behavior of A::foo, as projects might need

Reply:
fixed

Reply:
thanks, I did not realize this. Good to know